File: app/src/main/java/me/zipi/navitotesla/model/TeslaApiResponse.java
package me.zipi.navitotesla.model;

import com.google.gson.annotations.SerializedName;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class TeslaApiResponse {

    @AllArgsConstructor
    @NoArgsConstructor
    @Data
    public static class ListType<T> {
        java.util.List<T> response;
        Integer count;
        String error;
        @SerializedName("error_description")
        String errorDescription;
    }

    @AllArgsConstructor
    @NoArgsConstructor
    @Data
    public static class ObjectType<T> {
        T response;
        Integer count;
        String error;
        @SerializedName("error_description")
        String errorDescription;
    }

    @AllArgsConstructor
    @NoArgsConstructor
    @Data
    public static class Result {
        Boolean result;
        Boolean queued;
    }
}

File: app/src/main/kotlin/me/zipi/navitotesla/model/TeslaApiResponse.kt
package me.zipi.navitotesla.model

import com.google.gson.annotations.SerializedName

class TeslaApiResponse {
    data class ListType<T>(
        var response: List<T> = listOf(),
        var count: Int = 0,
        var error: String? = null,

        @SerializedName("error_description")
        var errorDescription: String? = null,
    )


    class ObjectType<T>(
        var response: T?,
        var count: Int = 0,
        var error: String? = null,

        @SerializedName("error_description")
        var errorDescription: String? = null,
    )

    data class Result(
        var result: Boolean?,
        var queued: Boolean? = null,
    )
}